improver/calibration/reliability_calibration.py

```diff
@@ -0,0 +1,412 @@
+        if n_cycle_hours != 1 or n_forecast_periods != 1:
+            raise ValueError('Forecasts have been provided from differing '
+                             'cycle hours or forecast periods; these should '
+                             'be consistent between forecasts.')
+
+    def _create_cycle_hour_coord(self, forecast_reference_time):
+        """
+        Constructs a coordinate that contains the cycle hour for which the
+        reliability table is valid.
+
+        Args:
+            forecast_reference_time (iris.coord.DimCoord):
+                The forecast_reference_time coordinate to be uses in the
+                coordinate creation.
+        Returns:
+            iris.coord.DimCoord:
+                A dimension coordinate containing an integer unitless
+                representation of the cycle hour.
+        """
+        cycle_hour, = self._get_cycle_hours(forecast_reference_time)
+        cycle_coord = iris.coords.DimCoord(cycle_hour, long_name='cycle_hour',
+                                           units=1)
+        return cycle_coord
+
+    @staticmethod
+    def _define_metadata(forecast_slice, diagnostic):
+        """
+        Define metadata that is specifically required for reliability table
+        cubes, whilst ensuring any mandatory attributes are also populated.
+
+        Args:
+            forecast_slice (iris.cube.Cube):
+                The source cube from which to get pre-existing metadata of use.
+            diagnostic (str):
+                The name of the diagnostic within the cube.
+        Returns:
+            dict:
+                A dictionary of attributes that are appropriate for the
+                reliability table cube.
+        """
+        attributes = generate_mandatory_attributes([forecast_slice])
+        attributes["title"] = "Reliability calibration data table"
+        attributes["diagnostic_standard_name"] = diagnostic
+        return attributes
+
+    def _create_reliability_table_cube(self, forecast, threshold_coord):
+        """
+        Construct a reliability table cube and populate it with the provided
+        data.
+
+        Args:
+            forecast (iris.cube.Cube):
+                A cube slice across the spatial dimensions of the forecast
+                data. This slice provides the time and threshold values that
+                relate to the reliability_table_data.
+            threshold_coord (iris.coords.DimCoord):
+                The threshold coordinate.
+        Returns:
+            iris.cube.Cube:
+                A reliability table cube.
+        """
+        def _get_coords_and_dims(coord_names):
+            coords_and_dims = []
+            leading_coords = [probability_bins_coord, reliability_index_coord]
+            for coord_name in coord_names:
+                try:
+                    crd = forecast_slice.coord(coord_name)
+                except CoordinateNotFoundError as err:
```

Comment:
Is it correct that this exception is only ever going to catch a missing `forecast_period` coordinate? That seems to be tha case given that the spatial coordinates have been taken from the `forecast` cube, so they can't be missing from the `forecast_slice` cube, and the `diagnostic` coordinate is also already checked for on the `forecast` cube. Do you think that perhaps it's not necessary to catch and raise and exception message here?

Reply:
Removed. The check_forecast_consistency call will already raise an error if there is no forecast_period coordinate. Good spot.